Translate __FILE__ and __LINE__

jdub opened the pull request:
- Stop preprocessor from defining **FILE** and **LINE**
- Translate them to file!().as_ptr() and line!() respectively

jdub committed:
Translate __FILE__ and __LINE__

- Stop preprocessor from defining __FILE__ and __LINE__
- Translate them to file!().as_ptr() and line!() respectively

jdub commented:
Because there are no parameters to these functions, the (necessary?) parameter list member doesn't matter, does it? I can't pass an empty list?

jameysharp commented:
The reason you're confused about what types to use as function arguments is that these aren't functions. :smile:

`__FILE__` should have type `IsPtr Rust.Immutable charType` and `__LINE__` should have type `IsInt Unsigned (BitWidth 32)`. (Technically the latter should probably be `Signed` but I suppose that might lead to a type mismatch with the corresponding Rust macro, and I don't think anyone will care.) In other words, exactly the types you used as the function return types.

Once that's fixed, I think this is spot on!

jdub committed:
Correct type signatures for file!().as_ptr() and line!()

I misunderstood what these type signatures were for: Although we
read file!().as_ptr() as a function, its meaning in the code, what
it represents, is a pointer to an array of characters. line!() on
the other hand is just a macro that expands to a u32.
